kartothek/io/eager_cube.py

```diff
@@ -419,7 +421,95 @@ def delete_cube(cube, store, datasets=None):
         store.delete(k)
 
 
-def copy_cube(cube, src_store, tgt_store, overwrite=False, datasets=None):
+def _transform_key(
+    src_key: str,
+    cube_prefix: str,
+    renamed_cube_prefix: Optional[str],
+    renamed_datasets: Optional[Dict[str, str]],
+):
+    """
+    Transform a key from <old cube prefix>++<old dataset><path> to
+    <new cube prefix>++<new dataset><path>
+
+    :param src_key:
+        Key to transform
+    :param cube_prefix:
+        Cube prefix before renaming
+    :param renamed_cube:
+        Optional new cube prefix
+    :param renamed_datasets:
+        Optional dict of {old dataset name: new dataset name} entries to rename datasets
+    """
+
+    tgt_key = src_key
+    if renamed_cube_prefix:
+        tgt_key = src_key.replace(
+            f"{cube_prefix}{KTK_CUBE_UUID_SEPARATOR}",
+            f"{renamed_cube_prefix}{KTK_CUBE_UUID_SEPARATOR}",
+        )
+
+    if renamed_datasets:
+        for ds_old, ds_new in renamed_datasets.items():
+            if f"{KTK_CUBE_UUID_SEPARATOR}{ds_old}" in tgt_key:
+                tgt_key = tgt_key.replace(
+                    f"{KTK_CUBE_UUID_SEPARATOR}{ds_old}",
+                    f"{KTK_CUBE_UUID_SEPARATOR}{ds_new}",
+                )
+    return tgt_key
+
+
+def _transform_metadata(
+    cube_prefix: str,
+    src_store: KeyValueStore,
+    renamed_cube: Optional[str] = None,
+    renamed_datasets: Optional[Dict[str, str]] = None,
+):
+    """
+    Create a dict of transformed metadata which contains the renamed cube and datasets.
+
+    :param cube_prefix:
+        Cube prefix before renaming
+    :param src_store:
+        Source Store (used to retrieve source datasets)
+    :param renamed_cube:
+        Optional new cube prefix
+    :param renamed_datasets:
+        Optional dict of {old dataset name: new dataset name} entries to rename datasets
+    """
+    all_datasets = discover_datasets_unchecked(
+        uuid_prefix=cube_prefix, store=src_store,
+    )
+    md_transformed = {}
+    tgt_cube_prefix = renamed_cube or cube_prefix
+    for ds, ds_meta in all_datasets.items():
+        # build new cube uuid for each dataset: <new prefix>++<new dataset name>
+        if (renamed_datasets is not None) and (ds in renamed_datasets.keys()):
+            tgt_uuid = (
+                f"{tgt_cube_prefix}{KTK_CUBE_UUID_SEPARATOR}{renamed_datasets[ds]}"
+            )
+        else:
+            tgt_uuid = f"{tgt_cube_prefix}{KTK_CUBE_UUID_SEPARATOR}{ds}"
+
+        # transform metadata for this dataset
+        md_ds_transformed = (
+            DatasetMetadataBuilder.from_dataset(ds_meta)
+            .modify_uuid(tgt_uuid)
+            .to_json()[1]
+        )
+        md_ds_key = f"{cube_prefix}{KTK_CUBE_UUID_SEPARATOR}{ds}{METADATA_BASE_SUFFIX}{METADATA_FORMAT_JSON}"
```

Comment:
This looks suspicious. At this place, there should not be the need to specify a `KTK_CUBE_UUID_SEPARATOR`. The cube separator is part of the dataset UUID and I would not introduce a special treatment like this here

Reply:
Fixed.